docs/guides/database.rst

```diff
@@ -193,9 +215,21 @@ $type
 $subtype
 	Entity subtype, e.g. ``'blog'`` or ``'page'``.
 
-By convention entities that have an uploaded avatar or icon will have the ``icontime`` property
-assigned. This means that you can use ``$entity->icontime`` to check if an icon exists for the given
-entity.
+Icons are assigned a timestamp when they're created. You can use ``$object->getIconTs($size)`` to 
+get the timestamp if you need to track changes to the icon.
+
+To check if an icon is set, use ``$object->hasIcon($size)``.
+
+Icon methods support passing an icon type if an entity has more than one icon. For example, a user
+might have an avatar and a cover photo icon. You would pass ``'cover_photo'`` as the icon type:
+
+.. code:: php
+	$object->setIconFromUploadedFile('uploaded_photo', 'cover_photo');
+
+	$object->getIconUrl([
+		'size' => 'medium',
+		'type' => 'cover_photo'
```

Comment:
Ok. I will leave you to it. Sorry for all nagging. Just want us to get it
right :)
On Apr 16, 2015 5:50 PM, "Brett Profitt" <notifications@github.com> wrote:

> In docs/guides/database.rst
> <https://github.com/Elgg/Elgg/pull/8202#discussion_r28522012>:
>
> > -assigned. This means that you can use ``$entity->icontime`` to check if an icon exists for the given
> > -entity.
> > +Icons are assigned a timestamp when they're created. You can use ``$object->getIconTs($size)`` to
> > +get the timestamp if you need to track changes to the icon.
> > +
> > +To check if an icon is set, use ``$object->hasIcon($size)``.
> > +
> > +Icon methods support passing an icon type if an entity has more than one icon. For example, a user
> > +might have an avatar and a cover photo icon. You would pass ``'cover_photo'`` as the icon type:
> > +
> > +.. code:: php
> > +	$object->setIconFromUploadedFile('uploaded_photo', 'cover_photo');
> > +
> > +	$object->getIconUrl([
> > +		'size' => 'medium',
> > +		'type' => 'cover_photo'
>
> Yes, we'd need to cache it like we do with viewtypes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8202/files#r28522012>.
>